src/components/number-picker/number-picker.component.ts

```diff
@@ -83,19 +98,29 @@ export class NumberPickerComponent implements ControlValueAccessor {
         return this.id + '-input';
     }
 
-    increment(event: MouseEvent | KeyboardEvent): void {
-        event.preventDefault();
+    increment(event?: MouseEvent | KeyboardEvent): void {
+        if (event) {
+            event.preventDefault();
+        }
 
         if (!this.disabled) {
             this.value = Math.max(Math.min(this.value + this.step, this.max), this.min);
+
+            // account for javascripts terrible handling of floating point numbers
+            this.value = parseFloat(this.value.toPrecision(6));
```

Comment:
Should this make use of the `precision` input?

Reply:
Thanks! Well spotted. The issue should now be fixed